build_defs/checksums.build_defs

```diff
@@ -0,0 +1,220 @@
+"""
+A module containing a mapping of CPython interpreters to checksums.
+
+This is a generated file -- see //tools/checksum_fetcher
+"""
+
+SUPPORTED_VERSIONS = [
+    "3.8.12",
+    "3.8.13",
+    "3.8.14",
+    "3.8.15",
+    "3.8.16",
+    "3.9.10",
+    "3.9.11",
+    "3.9.12",
+    "3.9.13",
+    "3.9.14",
+    "3.9.15",
+    "3.9.16",
+    "3.10.2",
+    "3.10.3",
+    "3.10.4",
+    "3.10.5",
+    "3.10.6",
+    "3.10.7",
+    "3.10.8",
+    "3.10.9",
+    "3.11.1",
+]
+
+INTERPRETERS = {
+    "3.8.12": {
+        "20220227/cpython-3.8.12+20220227-aarch64-apple-darwin-install_only.tar.gz": "f9a3cbb81e0463d6615125964762d133387d561b226a30199f5b039b20f1d944",
```

Comment:
Like I said, I'm a little concerned about coming across as if we directly endorse these third party distributors. It's an established practice to have a `hashes` argument on the rule and just ask the user to provide the checksum for us. If we did that, we could probably use a format string to generate these URLs based on `CONFIG.OS` and `CONFIG.ARCH`. What do you think? 

We can then have a `CONFIG.PYTHON.TOOLCHAIN_URL` where users can choose to put their preferred provider. We could even include indygreg as the suggested option in the docs.

Reply:
I think I responded to most these concerns in my [other comment](https://github.com/please-build/python-rules/pull/88#issuecomment-1419206586).

I'll add that, for now, I avoided overcomplicating the rule with too many parameters. Some configuration is provided to tweak the defaults but otherwise it falls back to behaving like a `remote_file()`.

What do you think ?

Reply:
Also a `CONFIG.PYTHON.TOOLCHAIN_URL` is enticing. Unfortunately, since this rule relies on GitHub releases having a weird URL format, I'm not sure this is the best way to go about it.